File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03391/p03391_num6_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<int> A(N), B(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i] >> B[i];
    }
    
    long long candies = 0;
    for (int i = 0; i < N; i++) {
        candies += min(A[i], B[i]);
        A[i] -= min(A[i], B[i]);
        B[i] -= min(A[i], B[i]);
    }
    
    cout << candies << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03391/p03391_num6_parsed.py
# Implementation:

def optimal_candies(N, AB):
    candies = 0
    
    while True:
        # Check if A and B are equal sequences
        if all(x == AB[0][0] for x, _ in AB):
            break
        
        # Find the index of the maximum element in A
        max_A_idx = max(range(N), key=lambda i: AB[i][0])
        # Find the index of the minimum element in B
        min_B_idx = min(range(N), key=lambda i: AB[i][1])
        
        # Decrease A and B by 1 according to the optimal strategy
        AB[max_A_idx][0] -= 1
        AB[min_B_idx][1] -= 1
        
        # Give one candy to Takahashi
        candies += 1
    
    return candies

N = int(input())
AB = [list(map(int, input().split())) for _ in range(N)]

print(optimal_candies(N, AB))